XcodeAutoBasher/XcodeAutoBasher/Models/VOKScriptForFolder.m

```diff
@@ -155,10 +155,13 @@ - (void)actuallyRunScript
 {
     //Check if file is runnable or Xcode will crash
     if ([[NSFileManager defaultManager] isExecutableFileAtPath:self.absolutePathToScript]) {
-        NSTask *task = [[NSTask alloc] init];
-        task.launchPath = self.absolutePathToScript;
-        task.currentDirectoryPath = [task.launchPath stringByDeletingLastPathComponent];
-        [task launch];
+        dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
+            NSTask *task = [[NSTask alloc] init];
+            task.launchPath = self.absolutePathToScript;
+            task.currentDirectoryPath = [task.launchPath stringByDeletingLastPathComponent];
+            task.environment = self.containingProject.environmentVariables;
```

Comment:
Hmmm - this will make every single task launched always run on a background thread, which might not be expected behavior. 

What about dispatching async to get the environment variables then dispatching back to the main queue to launch the task?